.github/workflows/ci.yml

```diff
@@ -13,16 +13,16 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 5
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
```

Comment:
@mohammadualam out of interest is it best practice to pin by hash even for trusted workflow providers from github. IE the ones from github seem very unlikely to have a supply chain attack.

Reply:
It is best practice to pin it even if it is a trusted source since an upstream compromise can happen at any step, including within trusted sources. It is the established trust of an app that the attackers are hoping would make users be more relaxed about this practice of pinning.